### ISSUE TITLE
Improve debugging experience when debugging host != build machine

### ISSUE THREAD

**Issue by jschwe** [servo/mozjs#485]
The debug information for some of the files currently points to the target directory, including an unpredictable hash.
This makes it hard to setup the debugger to show the source code while debugging.

Example: 
```
(lldb) image lookup -vn IsCatchableExceptionStatus
[redacted]
      LineEntry: [0x0000005b879f25d4-0x0000005b879f25dc): /data/servo/target/aarch64-unknown-linux-ohos/debug/build/mozjs_sys-af0e41768545dd53/out/build/dist/include/js/Exception.h:80
```

This makes it hard to setup lldb's `settings set target.source-map /path/on/build-machine D:\path\on\debugger-host-machine`, since the hash will change frequently.
At first glance it seems like the header files are also present in the source directory, and identical, so it would be preferable, if the original files at the fixed location could be used when compiling mozjs, instead of the files in the target directory.

**Comment by sagudev** [comment]
I think that's the limitation of how SM build system works (creates `out/build` dir and works from there), I wonder how this is worked around in Firefox?

**Comment by jschwe** [comment]
Yeah, thats a bit unfortunate. I guess I should count myself lucky, that it only affects a couple headers in `dist/include`, while the source .cpp files seem to be mostly unaffected (based on an entirely unrepresentative sample set of my current debugging session)
